sr_object_symmetry/include/sr_object_symmetry/reflectional_symmetry_detection_core.hpp

```diff
@@ -0,0 +1,577 @@
+
+      // If this is the smallest error of fit so far - record it
+      if (symNormalfitError < minSymNormalFitError)
+      {
+        minSymNormalFitError = symNormalfitError;
+        bestMatchId = nbrId;
+      }
+    }
+
+    if (bestMatchId != -1)
+      correspondences.push_back(pcl::Correspondence(pointId, bestMatchId, minSymNormalFitError));
+  }
+
+  // Correspondence rejection one to one
+  pcl::registration::CorrespondenceRejectorOneToOne correspRejectOneToOne;
+  correspRejectOneToOne.getRemainingCorrespondences(correspondences, correspondences);
+
+  // Check if there are enough correspondences
+  if (correspondences.size() == 0)
+  {
+    //       std::cout << "[sym::refineReflSymPosition] not enough correspondences" << std::endl;
+    return false;
+  }
+
+  //------------------------------------------------------------------------
+  // Update symmetry postition
+
+  // Calculate median symmetry position offset
+  std::vector<float> positionFitErrors(correspondences.size());
+
+  for (size_t crspId = 0; crspId < correspondences.size(); crspId++)
+  {
+    int queryId = correspondences[crspId].index_query;
+    int matchId = correspondences[crspId].index_match;
+    positionFitErrors[crspId] = sym::getReflSymPositionFitError(cloud_ds->points[queryId].getVector3fMap(),
+                                                                cloud->points[matchId].getVector3fMap(), symmetry);
+  }
+
+  // Get the median offset
+  float medianPositionFitError = utl::median<float>(positionFitErrors);
+
+  // Update symmetry
+  symmetry_refined.setOrigin(symmetryOrigin + symmetryNormal * medianPositionFitError);
+
+  return true;
+}
+
+//----------------------------------------------------------------------------
+// Symmetry global refinement
+//----------------------------------------------------------------------------
+
+/** \brief A functor for refining a symmetry candidate in a
+   * Levenberg-Marquardt optimization. Given a set of symmetric correspondences
+   * between oriented points and an initial reflectional symmetry candidate,
+   * find a relfectional symmetry candidate that minimizes the point to plane
+   * distance between the first point corresponding point and the reflection
+   * of the second corresponding point.
+   * Symmetry parameters are stored in a 6 dimensional vector
+   * (symmetry origin, symmetry normal).
+   * \note the final refined symmetry vector may have a non-unit normal.
+   */
+template <typename PointT>
+struct ReflSymRefineFunctor : BaseFunctor<float>
+{
+  /** \brief Empty constructor */
+  ReflSymRefineFunctor() {}
+
+  /** \brief Compute fitness for each input point.
+     *  \param[in]  x coefficients of the symmetry plane
+     *  \param[out] fvec error vector
+     */
+  int operator()(const Eigen::VectorXf& x, Eigen::VectorXf& fvec) const
+  {
+    // Get current rotational symmetry
+    ReflectionalSymmetry symmetry(x.head(3), x.tail(3));
+
+    //       // Compute occlusion scores
```

Comment:
Can we remove this commented out section?

Reply:
done